src/util/HeaderUtil.ts

```diff
@@ -226,13 +224,14 @@ function parseAcceptPart(part: string, replacements: Record<string, string>): Ac
     if (name === 'q') {
       // Extension parameters appear after the q value
       map = extensionParams;
-      testQValue(value);
-      weight = Number.parseFloat(value);
+      if (isValidQValue(value)) {
```

Comment:
I think we need to distinguish between _valid_ q-values and _parseable_ q-values.
For example, `0.3333` is technically an invalid value that the client, according to RFC 7231, MUST NOT send. However, whether this value is still parsed is at the discretion of the server.

I think the more helpful thing to do there, as per the robustness principle, is to accept anything that looks like `0.000…`, `1.000…` and all decimal numbers in between.